client/json-rpc/rpc_procedures.c

```diff
@@ -0,0 +1,328 @@
+	cJSON_AddItemToObject(item, "balance", balance);
+	cJSON_AddItemToObject(item, "key", key);
+	
+	cJSON_AddItemToArray(d->root, item);
+	return 0;
+}
+
+cJSON * method_xdag_get_account(struct xdag_rpc_context *ctx, cJSON *params, cJSON *id, char *version);
+cJSON * method_xdag_get_account(struct xdag_rpc_context *ctx, cJSON *params, cJSON *id, char *version)
+{
+	xdag_debug("rpc call method get_account, version %s",version);
+	struct rpc_account_callback_data cbdata;
+	cbdata.count = (g_is_miner ? 1 : 20);
+	if (params) {
+		if (cJSON_IsArray(params)) {
+			size_t size = cJSON_GetArraySize(params);
+			int i = 0;
+			for (i = 0; i < size; i++) {
+				cJSON* item = cJSON_GetArrayItem(params, i);
+				if(cJSON_IsString(item)) {
+					cbdata.count = atoi(item->valuestring);
+					break;
+				}
+			}
+		} else {
+			ctx->error_code = 1;
+			ctx->error_message = strdup("Invalid parameters.");
+		}
+	}
+		
+	cJSON * ret = NULL;
+	if(ctx->error_code == 0) {
+		if(g_xdag_state < XDAG_STATE_XFER) {
+			ctx->error_code = 1;
+			ctx->error_message = strdup("Not ready to show a balance.");
+		} else {
+			ret = cJSON_CreateArray();
+			cbdata.root = ret;
+			xdag_traverse_our_blocks(&cbdata, &rpc_account_callback);
+		}
+	}
+	
+	return ret;
+}
+
+
+/* balance */
+/*
+ request:
+ "method":"get_balance", "params":["A"], "id":1
+ "jsonrpc":"2.0", "method":"account", "params":["A"], "id":1
+ "version":"1.1", "method":"account", "params":["A"], "id":1
+ 
+ reponse:
+ "result":[{"balance":"10.111111"}], "error":null, "id":1
+ "jsonrpc":"2.0", "result":[{"balance":"10.111111"}], "error":null, "id":1
+ "version":"1.1", "result":[{"balance":"10.111111"}], "error":null, "id":1
+ */
+cJSON * method_xdag_get_balance(struct xdag_rpc_context * ctx, cJSON * params, cJSON *id, char *version);
+cJSON * method_xdag_get_balance(struct xdag_rpc_context * ctx, cJSON * params, cJSON *id, char *version)
+{
+	xdag_debug("rpc call method get_balance, version %s", version);
+	char address[128] = {0};
+	if (params) {
+		if (cJSON_IsArray(params)) {
+			size_t size = cJSON_GetArraySize(params);
+			int i = 0;
+			for (i = 0; i < size; i++) {
+				cJSON* item = cJSON_GetArrayItem(params, i);
+				if(cJSON_IsString(item)) {
+					strcpy(address, item->valuestring);
+					break;
+				}
+			}
+		} else {
+			ctx->error_code = 1;
+			ctx->error_message = strdup("Invalid parameters.");
+			return NULL;
+		}
+	}
+	
+	if(g_xdag_state < XDAG_STATE_XFER) {
+		ctx->error_code = 1;
+		ctx->error_message = strdup("Not ready to show a balance.");
+		return NULL;
+	} else {
+		cJSON * ret = NULL;
+		cJSON* item = cJSON_CreateObject();
+		xdag_hash_t hash;
+		xdag_amount_t balance;
+		
+		if(strlen(address)) {
+			xdag_address2hash(address, hash);
+			balance = xdag_get_balance(hash);
+		} else {
+			balance = xdag_get_balance(0);
+		}
+		
+		char str[128] = {0};
+		sprintf(str, "%.9Lf",  xdag_amount2xdag(balance) + (long double)xdag_amount2cheato(balance) / 1000000000);
```

Comment:
You moved declaration of amount2xdags to the header file. You could use existing function 
long double amount2xdags(xdag_amount_t amount)
{
	return xdag_amount2xdag(amount) + (long double)xdag_amount2cheato(amount) / 1000000000;
}

Reply:
Good point!

Reply:
Will you fix this line?